extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -82,11 +83,21 @@ BsonDiscriminatorBuildItem collectBsonDiscriminators(CombinedIndexBuildItem inde
     }
 
     @BuildStep
-    List<ReflectiveClassBuildItem> addCodecsAndDiscriminatorsToNative(CodecProviderBuildItem codecProviders,
-            BsonDiscriminatorBuildItem bsonDiscriminators) {
+    CommandListenerBuildItem collectCommandListeners(CombinedIndexBuildItem indexBuildItem) {
+        Collection<ClassInfo> commandListenerClasses = indexBuildItem.getIndex()
+                .getAllKnownImplementors(DotName.createSimple(CommandListener.class.getName()));
+        List<String> names = commandListenerClasses.stream().map(ci -> ci.name().toString()).collect(Collectors.toList());
+        return new CommandListenerBuildItem(names);
+    }
+
+    @BuildStep
+    List<ReflectiveClassBuildItem> addCodecsAndDiscriminatorsAndListenersToNative(CodecProviderBuildItem codecProviders,
```

Comment:
Maybe rename the method to `addExtensionPointsToNative` ?

Reply:
Sure